articles/060_ros_middleware_interface.md

```diff
@@ -170,44 +171,19 @@ However the performance using the DynamicData API will likely always be lower co
 
-To avoid the overhead of rebuilding everything a different set of binary packages could be provided.
+To avoid the need to build from source a set of binary packages could be provided which chose the middleware implementation at build time.
 While this would reduce the effort for the user the buildfarm would need to build a completely separate set of binary packages.
-The effort to support N packages with M different middleware implementation would require significant resources for maintaining the service as well as the necessary computing power.
-
-#### Reduce the number of middleware implementation specific packages
-
-One way to at least reduce the effort for building for different middleware implementations is to reduce the number of packages depending on the specific middleware implementation.
-This can, for example, be achieved using the DynamicData API mentioned before.
-Since only a few packages need to be built for each middleware implementation it would be feasible to generate binary packages for them on the buildfarm.
-
-The user could then install (one | a few) binary package(s) for a specific middleware implementation together with its specific implementation of the mapping between the message specification and the DynamicData API.
-All other binary packages would be agnostic to the selected middleware implementation.
+The effort to support N packages with M different middleware implementation would require significant resources (M * N) for maintaining the service as well as the necessary computing power.
 
-            +-----------+
-            | user land |
-            +-----------+
-                  |
-                  v
-            +-----------+           Generic binary packages
-            | msg pkgs  |           agnostic to selected
-            +-----------+           middleware implementation
-                  |
-                  ?                 Select middleware implementation
-                /   \               by installing (one | a few)
-              /       \             binary package(s)
-    +-----------+   +-----------+
-    | mw impl 1 |   | mw impl 2 |
-    +-----------+   +-----------+
+#### Decide at runtime
 
-#### Generate "fat" binary packages
-
-Another way to enable the user to switch between middleware implementations without the need to use the DynamicData API is to embed the middleware specific generated code for all supported implementations into each binary package.
-The specific middleware implementation would then be selected, for example, at link time and only the corresponding generated code of that middleware implementation would be used.
-
-### Using single middleware implementation only
+The alternative is to support selecting a specific middleware implementation at runtime.
+This will require that the to be selected middleware implementation was available at compile time in order for the middleware specific type support to be generated for every message package.
```

Comment:
this requires ? that's how our current code base behaves right?

Reply:
Indeed. Fixed in 104e93ba9da14e04dba2e60ad4ea96a5e5ba09d0.